jni/src/faiss_wrapper.cpp

```diff
@@ -684,20 +684,29 @@ jobjectArray knn_jni::faiss_wrapper::QueryBinaryIndex_WithFilter(knn_jni::JNIUti
     } else {
         faiss::SearchParameters *searchParameters = nullptr;
         faiss::SearchParametersHNSW hnswParams;
+        faiss::SearchParametersIVF ivfParams;
         std::unique_ptr<faiss::IDGrouperBitmap> idGrouper;
         std::vector<uint64_t> idGrouperBitmap;
-        auto hnswReader = dynamic_cast<const faiss::IndexBinaryHNSW*>(indexReader->index);
+        auto ivfReader = dynamic_cast<const faiss::IndexBinaryIVF*>(indexReader->index);
         // TODO currently, search parameter is not supported in binary index
         // To avoid test failure, we skip setting ef search when methodPramsJ is null temporary
-        if(hnswReader!= nullptr && (methodParamsJ != nullptr || parentIdsJ != nullptr)) {
-            // Query param efsearch supersedes ef_search provided during index setting.
-            hnswParams.efSearch = knn_jni::commons::getIntegerMethodParameter(env, jniUtil, methodParams, EF_SEARCH, hnswReader->hnsw.efSearch);
-            if (parentIdsJ != nullptr) {
-                idGrouper = buildIDGrouperBitmap(jniUtil, env, parentIdsJ, &idGrouperBitmap);
-                hnswParams.grp = idGrouper.get();
+        if (ivfReader) {
```

Comment:
might be a good idea to throw in a unit test in this or as an immediate follow up

Reply:
Sure, will do in follow-up